### ISSUE TITLE
Custom Field Logic Bypass

### ISSUE THREAD

**Issue by deckar01** [marshmallow-code/marshmallow#2012]
If you want to create a field that transforms `None` to a preset value, a custom field is required.

Currently the documentation for "Creating A Field Class" states:

> To create a custom field class, create a subclass of [marshmallow.fields.Field](https://marshmallow.readthedocs.io/en/latest/marshmallow.fields.html#marshmallow.fields.Field) and implement its [_serialize](https://marshmallow.readthedocs.io/en/latest/marshmallow.fields.html#marshmallow.fields.Field._serialize) and/or [_deserialize](https://marshmallow.readthedocs.io/en/latest/marshmallow.fields.html#marshmallow.fields.Field._deserialize) methods.

https://marshmallow.readthedocs.io/en/latest/custom_fields.html#creating-a-field-class

```py
class Integer(fields.Integer):
    def _deserialize(self, value, *args, **kwargs):
        value = value or 0
        return super()._deserialize(value, *args, **kwargs)

class Test(Schema):
    foo = Integer(allow_none=True)

Test().load({'foo': None}) # but it produces {"foo": None} instead of {"foo": 0}
```

When `None` (or `missing`) is encountered `_deserialize()` is never executed, because `Field.deserialize()` returns immediately. The current work around appears to require an understanding of undocumented parts of the `Field` implementation and extending `deserialize()` instead. The documentation should probably explain the `_/de/serialize()` and None/missing interaction.

```py
class Amount(fields.Integer):
    def deserialize(self, value, *args, **kwargs):
        value = value or 0
        return super(Amount, self).deserialize(value, *args, **kwargs)
```

A similar use case is having a `Nested` field supply the default values for the nested schema's fields when the parent field is `None`/missing. We may want to explore improving the default behavior in a future version, but for now providing a pattern for this in the docs would be helpful.

```py
class Options(fields.Nested):
    def deserialize(self, value, *args, **kwargs):
        value = value or {}
        return super(Options, self).deserialize(value, *args, **kwargs)
```

Related to #388

**Comment by ThiefMaster** [comment]
Being able to use the `load_default` logic when receiving a `None` value would probably already cover many usecases...

**Comment by lafrech** [comment]
> Being able to use the `load_default` logic when receiving a `None` value would probably already cover many usecases...

I believe this is the point of https://github.com/marshmallow-code/marshmallow/pull/1381 (from https://github.com/marshmallow-code/marshmallow/issues/713#issuecomment-529154890).

**Comment by ThiefMaster** [comment]
Yes, this would work perfectly fine for me! Any chance this will make it into a release soon-ish? :)

**Comment by lafrech** [comment]
> Any chance this will make it into a release soon-ish? :)

I won't be working on it in the coming weeks. If anyone wants to take over:
- rework to `load_default_values`
- add symmetrical `dump_default_values`
- update `__repr__`
- docs
